.yarn/patches/lockfile-lint-api-npm-5.4.6-dc86b73900.patch

```diff
@@ -0,0 +1,47 @@
+diff --git a/src/ParseLockfile.js b/src/ParseLockfile.js
+index 0f0c951027ec83c61769bb6a48943420dff133b8..bad2d251cf376bf3ef4b444a0d49f03a602d7a6e 100644
+--- a/src/ParseLockfile.js
++++ b/src/ParseLockfile.js
+@@ -21,13 +21,13 @@ const {
+  * @return boolean
+  */
+ function checkSampleContent (lockfile) {
+-  const [sampleKey, sampleValue] = Object.entries(lockfile)[0]
++  const [sampleKey, sampleValue] = Object.entries(lockfile)[1]
+   return (
+     sampleKey.match(/.*@.*/) &&
+     (sampleValue &&
+       typeof sampleValue === 'object' &&
+       sampleValue.hasOwnProperty('version') &&
+-      sampleValue.hasOwnProperty('resolved'))
++      sampleValue.hasOwnProperty('resolution'))
+   )
+ }
+ /**
+@@ -41,7 +41,25 @@ function yarnParseAndVerify (lockfileBuffer) {
+   if (!hasSensibleContent) {
+     throw Error('Lockfile does not seem to contain a valid dependency list')
+   }
+-  return {type: 'success', object: lockfile}
++  const normalized = Object.fromEntries(Object.entries(lockfile).map(([packageName, packageDetails]) => {
++    const resolution = packageDetails.resolution;
++    if (!resolution) {
++      return [packageName, packageDetails];
++    }
++    const splitByAt = resolution.split('@');
++    let [resolvedPackageName, host] = splitByAt;
++    if (splitByAt.length > 2) {
++      resolvedPackageName = `${splitByAt[0]}${splitByAt[1]}`;
++      host = splitByAt[2];
++    }
++
++    if (splitByAt.length > 2 && resolution[0] !== '@') {
++      [resolvedPackageName, host] = splitByAt;
++    }
```

Comment:
In addition to the above, should it `${splitByAt[0]}@${splitByAt[1]}`, not `${splitByAt[0]}${splitByAt[1]}`? (because in the case where the package name is scoped, `splitByAt[0]` _should_ be an empty string)

Reply:
@mcmire so in the case of splitByAt > 2 && resolution[0] === '@' the splitByAt[0] is '' because the '@' is at the beginning of the string. So the splitByAt[0] is not needed and instead should be replaced by '@' which I did.